### PR TITLE
feat: add template for local pubsub using python sdk

### PR DESCRIPTION
**Pull request by saranyailla**
**Issue #, if available:**

**Description of changes:**
Add a new template for local communication between greengrass components over IPC. 
**Why is this change necessary:**

**How was this change tested:**

**Any additional information or context required to review the change:**

**Checklist:**
- [ ] Updated the README if applicable
- [ ] Updated or added new unit tests
- [ ] Updated or added new integration tests
- [ ] Updated or added new end-to-end tests
- [ ] If your code makes a remote network call, it was tested with a proxy
- [ ] You confirm that the change is backwards compatible

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: templates/python/LocalPubsub/README.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
+## My Project
```
Comment (Nelsonochoam) [inline templates/python/LocalPubsub/README.md:1]: Maybe add a description here on how to install it and run the tests or what this templated does

File: templates/python/LocalPubsub/localpubsub/pub.py [hunk 2]
```diff
@@ -0,0 +1,14 @@
+from awsiot.greengrasscoreipc.clientv2 import GreengrassCoreIPCClientV2
```
Comment (Nelsonochoam) [inline templates/python/LocalPubsub/localpubsub/pub.py:1]: `publisher.py`?

File: templates/python/LocalPubsub/localpubsub/sub.py [hunk 3]
```diff
@@ -0,0 +1,31 @@
+import sys
```
Comment (Nelsonochoam) [inline templates/python/LocalPubsub/localpubsub/sub.py:1]: `subscriber.py`?

File: templates/python/LocalPubsub/localpubsub/pub.py [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+
+def publish_binary_message_to_topic(ipc_client: GreengrassCoreIPCClientV2, topic, message):
+    binary_message = BinaryMessage(message=bytes(message, "utf-8"))
+    publish_message = PublishMessage(binary_message=binary_message)
+    return ipc_client.publish_to_topic(topic=topic, publish_message=publish_message)
+
+
+def publish_message_N_times(ipc_client, topic, message, N=10):
+    for i in range(1, N + 1):
+        print("Successfully published " + str(i) + " message(s)")
```
Comment (Nelsonochoam) [inline templates/python/LocalPubsub/localpubsub/pub.py:13]:
```suggestion
        publish_binary_message_to_topic(ipc_client, topic, message)
        print("Successfully published " + str(i) + " message(s)")
```

File: templates/python/LocalPubsub/test_LocalPubSub/test_pub.py [hunk 5]
```diff
@@ -0,0 +1,22 @@
+import pytest
```
Comment (Nelsonochoam) [inline templates/python/LocalPubsub/test_LocalPubSub/test_pub.py:1]: `test`? why the `tests_LocalPubsub` folder name? Following [PEP8 conversions](https://peps.python.org/pep-0008/#:~:text=of%20PEP%203131.-,package%20and%20module%20names,-Modules%20should%20have) it should be all lower case

File: templates/python/LocalPubsub/test_LocalPubSub/test_pub.py [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+    mocker.patch("awsiot.greengrasscoreipc.connect")
+    from awsiot.greengrasscoreipc.clientv2 import GreengrassCoreIPCClientV2
+
+    client = GreengrassCoreIPCClientV2()
+    return client
+
+
+def test_publish_message_N_times(ipc_client):
+    pub.publish_message_N_times(ipc_client, "topic", "World")
+    assert ipc_client.publish_to_topic.call_count == 10
```
Comment (Nelsonochoam) [inline templates/python/LocalPubsub/test_LocalPubSub/test_pub.py:17]: We should strive to make this sample test a bit better. Asserting call count doesn't tell us things are working. Maybe assert that it was called with what you expect

File: templates/python/LocalPubSub/test_localpubsub/test_publisher.py [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+    mocker.patch("awsiot.greengrasscoreipc.connect")
+    from awsiot.greengrasscoreipc.clientv2 import GreengrassCoreIPCClientV2
+
+    client = GreengrassCoreIPCClientV2()
+    return client
+
+
+def test_publish_message_N_times(ipc_client):
+    publisher.publish_message_N_times(ipc_client, "topic", "World")
+    assert ipc_client.publish_to_topic.call_count == 10
```
Comment (Nelsonochoam) [inline templates/python/LocalPubSub/test_localpubsub/test_publisher.py:17]: Don't assert the call count, check it gets called with the expected args at least.

**Review by Nelsonochoam — APPROVED** [review]

**Review by MikeDombo — APPROVED** [review]
LGTM

**Review by Nelsonochoam — APPROVED** [review]

**Pull request merged**